py/fiberassign/stucksky.py

```diff
@@ -0,0 +1,151 @@
+    
+def stuck_on_sky(hw, tiles):
+    '''
+    Will STUCK positioners land on good SKY locations for the given set of tiles?
+
+    Returns a nested dict:
+        stuck_sky[tileid][loc] = bool_good_sky
+    '''
+    import os
+    import numpy as np
+    from scipy.spatial import KDTree
+    import fitsio
+    from astropy.wcs import WCS
+    from fiberassign.hardware import FIBER_STATE_STUCK, FIBER_STATE_BROKEN
+    from fiberassign.utils import Logger
+
+    log = Logger.get()
+    skybricks_dir = os.environ.get('SKYBRICKS_DIR', None)
+    if skybricks_dir is None:
+        log.warning('Environment variable SKYBRICKS_DIR is not set; not looking up whether '
+                    'stuck positioners land on good sky')
+        return
+    skybricks_fn = os.path.join(skybricks_dir, 'skybricks-exist.fits')
+    skybricks = fitsio.read(skybricks_fn, upper=True)
+    skykd = _radec2kd(skybricks['RA'], skybricks['DEC'])
+
+    tilekd = _radec2kd(tiles.ra, tiles.dec)
+
+    # skybricks are 1 x 1 deg.
+    brickrad = (1. * np.sqrt(2.) / 2.)
+    searchrad = 1.01 * (hw.focalplane_radius_deg + brickrad)
+
+    # here, convert search radius to radians -- an overestimate vs
+    # unit-sphere distance, but that's the safe direction.
+    sky_indices = tilekd.query_ball_tree(skykd, np.deg2rad(searchrad))
+
+    stuck_sky = dict()
+    for tile_id, tile_ra, tile_dec, tile_theta, sky_inds in zip(
+            tiles.id, tiles.ra, tiles.dec, tiles.obstheta, sky_indices):
+        stuck_sky[tile_id] = dict()
+    
+        # Stuck locations and their angles
+        stuck_loc = [loc for loc in hw.locations
+                     if ((hw.state[loc] & FIBER_STATE_STUCK != 0) and
+                         (hw.state[loc] & FIBER_STATE_BROKEN == 0) and
+                         (hw.loc_device_type[loc] == 'POS'))]
+        stuck_theta = [hw.loc_theta_pos[x] for x in stuck_loc]
```

Comment:
These (loc_theta_pos, loc_phi_pos) coords should get offset by hw.loc_{theta,phi}_offset.  We do that in a later PR.